### DISCUSSION TITLE
Tested on centos' x86 platform, the 21st test case did not pass

### DISCUSSION THREAD

**Discussion by 123xyl** [openmopac/mopac#124] [Q&A]
![image](https://user-images.githubusercontent.com/96686347/194308426-447ff12f-580e-4a82-b600-825e3521eb90.png)
![image](https://user-images.githubusercontent.com/96686347/194309107-8d017427-610c-41fd-9eb9-b895e876c3f6.png)

**Comment by susilehtola** [comment]
Building software as root is a bad idea. Note also that MOPAC is available as a standard Fedora EPEL package, just run
`dnf -y install mopac`

**Comment by godotalgorithm** [comment]
Since nearly all of your tests passed, your version of MOPAC should be working correctly. I suspect that the one test is failing because your files aren't fully up to date with what is in the MOPAC repository. I can examine this failure further if you attach your versions of `tests/Compare_ADD-H_1EJG_and_1CBN.out` and `build/tests/Compare_ADD-H_1EJG_and_1CBN.out`. Otherwise, I don't have enough information to consider this problem any further.

**Reply by 123xyl** [reply]
My source files are version 22.0.3 not 22.0.4

**Reply by godotalgorithm** [reply]
I wouldn't worry about the failed test then, it can be explained by changes to the reference data for the tests between the two versions.

**Reply by 123xyl** [reply]
Using the multi-process running example, the same example is failed above, and the following is passed, is this situation passed, or did it not pass?
![image](https://user-images.githubusercontent.com/96686347/194525800-f42732bb-8bea-4b94-bf9d-a71e4ad69830.png)
![image](https://user-images.githubusercontent.com/96686347/194526082-d9cf8775-eaa4-4753-a7b5-dc9aac3516b3.png)

**Reply by 123xyl** [reply]
such as the 106th test case

**Reply by godotalgorithm** [reply]
I don't know exactly what you are doing here, but it looks like you are running two instances of CTest at the same time. When you do this, tests will randomly fail because they are sometimes trying to write to the same output file at the same time. You should only run one instance of CTest at a time. The tests in CTest can be run in parallel with the command `ctest -j num_proc` where `num_proc` is the number of simultaneous tests. The most recent version of MOPAC fixed some testing concurrency problems, so your version might have problems when `num_proc` is large.

**Reply by 123xyl** [reply]
ctest  with 16 processor

**Reply by 123xyl** [reply]
when I use  **ctest** ，all the test cases can be passed

**Reply by godotalgorithm** [reply]
What command are you using to run CTest with 16 processors?

**Reply by 123xyl** [reply]
mpirun --allow-run-as-root -np 16 ctest

**Reply by 123xyl** [reply]
mpirun --allow-run-as-root This is a tool for Huawei

**Reply by godotalgorithm** [reply]
Right, CTest is not MPI-enabled software. By using MPI (`mpirun`) to run CTest on 16 processors, you are running 16 independent versions of CTest that are all trying to write test outputs to the same files at the same time, which is causing random file collision errors. The only correct way to run CTest on 16 processes is `ctest -j 16` as CTest only supports threaded parallelism, not MPI parallelism.

**Reply by 123xyl** [reply]
ok thankyou

**Reply by 123xyl** [reply]
Do mopac test cases support multi-process operation and how to run it？

**Reply by 123xyl** [reply]
- port (Failed)
          2 - crambin (Failed)
          3 - mozyme (Failed)
          4 - xeno (Failed)
          6 - convert_edited_pdb (Failed)
          8 - ters_preserved (Failed)
          9 - compare_proteins (Failed)
         12 - pdb_compare (Failed)
         13 - pdb_compare2 (Failed)
         14 - pdb_compare3 (Failed)
         15 - pdb_compare4 (Failed)
         16 - pdb_compare5 (Failed)
         17 - pdb_compare6 (Failed)
         18 - add-h (Failed)
         19 - add-h2 (Failed)
         20 - add-h3 (Failed)
         21 - compare_add-h (Failed)
         22 - locate_ts (Failed)
         23 - ionize (Failed)
         24 - salt (Failed)
         25 - salt2 (Failed)
         26 - residues (Failed)
         38 - co_extern (Failed)
Errors while running CTest

**Reply by 123xyl** [reply]
![image](https://user-images.githubusercontent.com/96686347/194714425-7403d492-a496-42a0-a942-6b96d6d6ca6c.png)

**Reply by godotalgorithm** [reply]
Are you referring to multi-process operation of MOPAC itself? MOPAC is a serial program. Its only use of multi-processing at the moment is through threaded versions of BLAS and LAPACK libraries such as Intel MKL and OpenBLAS.
